lib/spack/spack/cmd/compiler.py

```diff
@@ -64,20 +64,14 @@ def setup_parser(subparser):
     # List
     list_parser = sp.add_parser("list", help="list available compilers")
     list_parser.add_argument(
-        "--scope",
-        action=arguments.ConfigScope,
-        default=lambda: spack.config.default_list_scope(),
-        help="configuration scope to read from",
+        "--scope", action=arguments.ConfigScope, help="configuration scope to read from"
```

Comment:
why no default?

Reply:
If not specified, the default is `None`, which for us means all scopes. Right now we are returning `None` in a very convoluted way, see https://github.com/spack/spack/pull/41741/commits/c3269c366512c019bcfd30a7cba499b6255ed2d5

Reply:
ah, missed that it returned None :laughing: